### DISCUSSION TITLE
Great project. ...How about X-Y mode, FFT, etc?

### DISCUSSION THREAD

**Discussion by adyn** [fhdm-dev/scoppy#51] [General]
Scoppy is a great project. I'm trying to help some science teachers and I'm building them some hardware (Pico + front end) so they can use them in class. As an educational tool this perfect. The overall cost is very affordable (unfortunately schools cannot afford even a cheap scope). It would be great if this would have some extra features:
* an X-Y mode
* FFT

Any plans for that?

**Comment by fhdm-dev** [comment]
Hi,

Yes, X-Y mode and FFT are both high on the list of features I'm planning to add to Scoppy. If there's anything else you need then let me know and I'll do what I can.

Regarding your front end, there's a few minor features in the next version that might influence your design. 

1. It's going to be a lot easier to change the GPIO assignments for the different functions (ie. logical analyzer, signal generator, voltage range pins, status LED). You'll be able to do this via cmake instead of having to modify the firmware source code itself.
2. The [voltage range pins](https://oscilloscope.fhdm.xyz/wiki/Analog-Front-Endl) can be used as outputs rather than inputs. This will allow the input gain to be switched automatically eg. via something like a cheap CD4052?

**Reply by adyn** [reply]
That is great news. Especially moving the input gain control into the software, from software to hardware and not as it is today (your point 2.). 

I think you should seriously consider designing a kit (PCB + components) for this or even an assembled product as this one:[ https://espotek.com/labrador]( https://espotek.com/labrador)/ 

I tried that product too. The hardware was quite good for my purpose (teaching / learning) but the software is not as polished as yours. Your Android App is replicating quite well a real oscilloscope and is very natural to somebody that used a standalone oscilloscope. It would work the other way too. If somebody would use your interface first and transition after that to an oscilloscope it would be very natural to that person.

There are other projects but with standalone solutions, like this one: [https://github.com/gustavollps/esp32TTGO_i2s_scope](https://github.com/gustavollps/esp32TTGO_i2s_scope) but these are not quite as good for education as the version using an App as the oscilloscope interface. That allows the teachers to cast the content on a TV/projector making it a better tool for class. Also a lot of students have smart phones that have way better screens than what you would be able to attach to a RPi, ESP32,etc. on a decent cost. 

I sent you an email if you are interested to expand this (iOS App, hardware kit, more features, etc.)
